test/AsteroidMining/RobotTest.java

```diff
@@ -0,0 +1,68 @@
+package test.AsteroidMining;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.BeforeEach;
+import src.AsteroidMining.*;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class RobotTest {
+
+    private Robot r;
+    private Asteroid a1, a2;
+
+    @BeforeEach
+    void init() {
+        r = new Robot();
+        a1 = new Asteroid(1, 1, new Carbon(), 2);
+        a2 = new Asteroid(1, 1, new Carbon(), 0);
+        a2.addNeighbour(a1);
+    }
+
+    @Test
+    public void testHideSuccess(){
+        r.setPlace(a2);
+        a2.removeResource();
+        assertTrue(a2.isHollow());
+        r.hide();
+        assertTrue(r.isHidden());
+    }
+
+    @Test
+    public void testHideNoSuccess(){
+        r.setPlace(a1);
+        assertFalse(a1.isHollow());
+        r.hide();
+        assertFalse(r.isHidden());
+    }
+
+    @Test
+    public void testDrill() {
+        r.setPlace(a1);
+        assertFalse(a1.isFullyDrilled());
+
+        r.drill();
+        assertTrue(a1.isFullyDrilled());
+    }
+
+    @Test
+    public void testFullyDrilled() {
+        r.setPlace(a2);
+        assertTrue(a2.isFullyDrilled());
+        //so it says it is not fully drilled even when depth is 0
+        assertFalse(r.drill());
+    }
```

Comment:
isFullyDrilledTest is present in AsteroidTest and it behaves correctly. I think there is no need to duplicate it here with assertTrue(a2.isFullyDrilled());

Reply:
Here is a snippet from `Visitor.java`

```
 public boolean drill(){
        System.out.println("Drilling!");
        Asteroid a1 = (Asteroid) this.getPlace();
        if(a1==null) return false;
        if(a1.depth>=0) {
            a1.deepenHole(2);
            return true;
        }else{
            System.out.println("Asteroid is fully drilled!");
            return false;}
    }
```

So here it will actually try to deepen the hole even if the depth is already 0 which I do not thing it is a correct behavior.

Reply:
So I guess the problem is that in the Visitor there is an additional validation that behaves wrongly (returns true), and this is why for Asteroid it works and not here.

Reply:
ack